### PR TITLE
ruby_gems - showing more info in the tiles

### PR DESCRIPTION
**Pull request by iambibhas**
Related to #248. 

Showing more info than just the name and description -

### CODE REVIEW

**Comment by dogweather** [comment]
:+1: Helps the user decide which to investigate further.

**Comment by moollaza** [comment]
@iambibhas this is great, thanks!

My only thought is that we should flip the downloads and license so that when the license isn't there we don't have blank space below the downloads.

/cc @abeyang -- this one is already live, we're just adding more info to the tiles

**Comment by iambibhas** [comment]
@moollaza already tried that, didn't help, as I've set `tileFooter: '2line'`. So if one div is absent, another goes to the top.

**Comment by moollaza** [comment]
@iambibhas nice fix. A slightly more graceful approach would be to set the `tileFooter` to `1line` when we have no Licenses? I think that will have the same effect, but I'm not sure.

**Comment by iambibhas** [comment]
@moollaza how to do that from inside the normalize function?

**Comment by moollaza** [comment]
@iambibhas my bad, I thought we could set the variant on a per-tile basis but it's not possible. We'll stick with your solution.

**Comment by abeyang** [comment]
Looks like it's getting there. How are we shortening the description? Is it by number of words? It seems to cut off a bit short.. can we increase it by just a bit?

pasted_image_4_2_15__10_01_am

Moreover, can we add the following css changes? Please add the proper namespace to it as well. :smile: 

```css
.tile__title {
  /* pull title up a bit */
  margin-top: -0.2em
}

.tile__sub {
  /* push desc down a bit */
  margin-bottom: 1em;
}

.has-foot .tile__content--lg {
  /* bring footer up a bit */
  height: 6.4em;
}
```

Finally, rather than `Licenses: MIT`, can we say `MIT License`? (Unless there can be multiple licenses...)

**Comment by iambibhas** [comment]
@abeyang I think the truncation is because of the use of `tileSnippet: 'large'`. I'm passing the whole text, but it's showing only that much.

The `licences` part of the API Response is an Array. So there is a chance that there might be more than one. But I'm yet to see any such example. 

I'll make the CSS changes and show you.

**Comment by abeyang** [comment]
@iambibhas you're right -- we'll work on the truncation on our end. 

Style changes are looking good! Can you add the class `tx-clr--grey-light` to the **subtitle** and the **download** lines? In the end, it should look something like this:

http://cl.ly/image/2Y3u381X1S2m

**Comment by iambibhas** [comment]
@abeyang I'm currently sending the subtitle from the normalize function. Not sure if I can add custom class to it. @moollaza @jagtalon can I?

**Comment by moollaza** [comment]
@iambibhas no it's not currently possible to add classes to those. We'll soon be able to specify colors and font-sizes from the `variants` obj but it's not yet implemented.

@abeyang can we implement the color changes later?

**Comment by abeyang** [comment]
@moollaza yup that's fine -- we'll just wait until we can apply it to all templates like this one.

**Comment by jagtalon** [comment]
@moollaza @abeyang looks like there's a bug with our template. Missing subtitles can make a tile small:

screen shot 2015-04-08 at 1 37 03 am

I'll make an internal issue for it.

**Comment by jagtalon** [comment]
@iambibhas `sourceUrl` gets double-encoded which leads to links such as https://rubygems.org/search?utf8=%E2%9C%93&query=coffee%2520script.

screen shot 2015-04-08 at 1 51 24 am

File: share/spice/ruby_gems/ruby_gems.js [hunk 1]
```diff
@@ -26,13 +26,42 @@
...
+            sort_default: 'downloads',
             normalize : function(item){
+                var licenses = [];
+
+                if (item.licenses) {
+                    var keys = Object.keys(item.licenses);
+
+                    for (var i = 0; i < keys.length; i++) {
+                        licenses.push(item.licenses[keys[i]]);
+                    };
```
Comment (jagtalon) [inline share/spice/ruby_gems/ruby_gems.js]: Looks like `licenses` is already an array in the result:

screen shot 2015-04-08 at 1 56 12 am

So all we need is:

```javascript
var licenses = item.licenses || [];
```

**Comment by jagtalon** [comment]
@abeyang since this is a built-in template, I'm a bit hesitant with using CSS. Do we need the CSS for all text templates since we're applying it here?

File: share/spice/ruby_gems/ruby_gems.js (hunk 1 above)
Comment (iambibhas) [inline share/spice/ruby_gems/ruby_gems.js]: When I last checked, it was a json object and not an array. Lemme check again.

Comment (iambibhas) [inline share/spice/ruby_gems/ruby_gems.js]: Huh. Looks like they changed the format since the last commit I made here. I'll update the spice.

**Comment by jagtalon** [comment]
@iambibhas got contacted by @abeyang. The CSS should go internally so you can `git rm` it from this repo. Other than that, if the issues above have been resolved, this should be good to go!

**Comment by moollaza** [comment]
>@moollaza @abeyang looks like there's a bug with our template. Missing subtitles can make a tile small:

For now, please set the subtitle to `&nbsp` (`''` or `' '` might work too?) when it doesn't exist. We need to make sure they're not undef so the template accounts for them. We'll fix this internally so this doesn't happen but that'll get this moving forward :)

**Comment by iambibhas** [comment]
@moollaza done.

**Comment by sdougbrown** [comment]
FYI, if you'd like to adjust the number of characters displayed in the snippet, you can use the new `snippetChars` property like so:
```
    meta: {
        ...
        snippetChars: 150
    }
```

Feel free to experiment with this to get some longer descriptions, if you like. :smile:

**Comment by jagtalon** [comment]
Noice. :+1: 

giphy

**Pull request merged**